python/Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py

```diff
@@ -0,0 +1,229 @@
+
+        assert path.isfile(path.join(getJobsPath(), 'cnt'))
+
+        assert path.isdir(getXMLDir(j))
+
+        assert path.isfile(getXMLFile(j))
+
+        assert path.isfile(getXMLFile(j) + '~')
+
+        assert path.isfile(getIndexFile(j))
+
+    def test_c_XMLAutoUpdated(self):
+
+        from Ganga.GPI import jobs
+
+        j=jobs(0)
+
+        XMLFileName = getXMLFile(j)
+
+        last_update = stat(XMLFileName)
+
+        j.name = testStr
+
+        from Ganga.Utility.Config import getConfig
+        flush_timeout = getConfig('Registry')['AutoFlusherWaitTime']
+        time.sleep(2.*flush_timeout)
+
+        newest_update = stat(XMLFileName)
+
+        assert newest_update.st_mtime > last_update.st_mtime
+
+
+    def test_d_XMLUpdated(self):
+
+        from Ganga.GPI import jobs, disableMonitoring
+
+        disableMonitoring()
+
+        j=jobs(0)
+
+        XMLFileName = getXMLFile(j)
+
+        last_update = stat(XMLFileName)
+
+        j.submit()
+
+        newest_update = stat(XMLFileName)
+
+        assert len(j.subjobs) == len(testArgs)
+
+        assert newest_update.st_mtime > last_update.st_mtime
+
+    def test_e_SubJobXMLExists(self):
+        """ Second get the job and check that getting it via jobs doesn't cause it to be loaded"""
+        from Ganga.GPI import jobs
+
+        assert len(jobs) == 1
+
+        j=jobs(0)
+
+        assert path.isdir(getXMLDir(j))
+
+        assert path.isfile(getSJXMLIndex(j))
+
+        for sj in j.subjobs:
+            XMLFileName = getSJXMLFile(sj)
+            assert path.isfile(XMLFileName)
+            assert path.isfile(XMLFileName+'~')
+
+    def test_f_testXMLContent(self):
+
+        from Ganga.Core.GangaRepository.VStreamer import to_file, from_file
+
+        from Ganga.GPI import jobs, Job
+        from Ganga.GPIDev.Base.Proxy import stripProxy
+
+        from tempfile import NamedTemporaryFile
+
+        j=jobs(0)
+        XMLFileName = getXMLFile(j)
+        assert path.isfile(XMLFileName)
+        handler = open(XMLFileName)
+        tmpobj, errs = from_file(handler)
+
+        assert hasattr(tmpobj, 'name')
+
+        assert tmpobj.name == testStr
+
+        new_temp_file = NamedTemporaryFile()
+
+        ignore_subs = ''
+
+        to_file(stripProxy(j), new_temp_file, ignore_subs)
+
+        new_temp_file2 = NamedTemporaryFile()
+
+        j2=Job()
+        j2.name=testStr
+
+        to_file(stripProxy(j2), new_temp_file2, ignore_subs)
+
+        import filecmp
+
+        assert filecmp.cmp(handler.name, new_temp_file.name)
+        assert not filecmp.cmp(new_temp_file.name, new_temp_file2.name)
+        handler.close()
+
+    def test_g_testSJXMLContent(self):
+
+        from Ganga.Core.GangaRepository.VStreamer import to_file, from_file
+
+        from Ganga.GPI import jobs
+        from tempfile import NamedTemporaryFile
+        from Ganga.GPIDev.Base.Proxy import stripProxy
+
+        counter = 0
+        for sj in jobs(0).subjobs:
```

Comment:
Probably, but I planned to add a assert test on the ```counter``` parameter and forgot to add it